server/src/test/java/org/elasticsearch/search/SearchHitsTests.java

```diff
@@ -21,88 +21,196 @@
+        assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+        SearchHits searchHits = SearchHits.fromXContent(parser);
+        assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+        assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+        return searchHits;
+    }
+
+    public void testToXContent() throws IOException {
+        SearchHit[] hits = new SearchHit[] {
+            new SearchHit(1, "id1", new Text("type"), Collections.emptyMap()),
+            new SearchHit(2, "id2", new Text("type"), Collections.emptyMap()) };
+
+        long totalHits = 1000;
+        float maxScore = 1.5f;
+        SearchHits searchHits = new SearchHits(hits, new TotalHits(totalHits, TotalHits.Relation.EQUAL_TO), maxScore);
+        XContentBuilder builder = JsonXContent.contentBuilder();
+        builder.startObject();
+        searchHits.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        builder.endObject();
+        assertEquals("{\"hits\":{\"total\":{\"value\":1000,\"relation\":\"eq\"},\"max_score\":1.5," +
+            "\"hits\":[{\"_type\":\"type\",\"_id\":\"id1\",\"_score\":null},"+
+            "{\"_type\":\"type\",\"_id\":\"id2\",\"_score\":null}]}}", Strings.toString(builder));
+    }
+
+    public void testFromXContentWithShards() throws IOException {
+        for (boolean withExplanation : new boolean[] {true, false}) {
+            final SearchHit[] hits = new SearchHit[]{
+                new SearchHit(1, "id1", new Text("type"), Collections.emptyMap()),
+                new SearchHit(2, "id2", new Text("type"), Collections.emptyMap()),
+                new SearchHit(10, "id10", new Text("type"), Collections.emptyMap())
+            };
+
+            for (SearchHit hit : hits) {
+                String index = randomAlphaOfLengthBetween(5, 10);
+                String clusterAlias = randomBoolean() ? null : randomAlphaOfLengthBetween(5, 10);
+                final SearchShardTarget shardTarget = new SearchShardTarget(randomAlphaOfLengthBetween(5, 10),
+                    new ShardId(new Index(index, randomAlphaOfLengthBetween(5, 10)), randomInt()), clusterAlias, OriginalIndices.NONE);
+                if (withExplanation) {
+                    hit.explanation(SearchHitTests.createExplanation(randomIntBetween(0, 5)));
+                }
+                hit.shard(shardTarget);
+            }
+
+            long totalHits = 1000;
+            float maxScore = 1.5f;
+            SearchHits searchHits = new SearchHits(hits, new TotalHits(totalHits, TotalHits.Relation.EQUAL_TO), maxScore);
+            for (XContentType xContentType : XContentType.values()) {
+                XContentBuilder builder = XContentBuilder.builder(xContentType.xContent());
+                builder.startObject();
+                searchHits.toXContent(builder, ToXContent.EMPTY_PARAMS);
+                builder.endObject();
+                final BytesReference bytes = BytesReference.bytes(builder);
```

Comment:
could we replace these last few lines with `XContentHelper.toXContent(searchHits, xContentType, false)` ? Or maybe even toShuffledXContent but I am not sure if that may cause problems.